caas/specs/legacy.go

```diff
@@ -11,24 +11,12 @@ import (
 // a pod on the CAAS substrate.
 type PodSpecLegacy struct {
 	podSpecBase `yaml:",inline"`
-
-	// legacy version has containers/initContainers two blocks.
-	InitContainers []ContainerSpec `json:"initContainers" yaml:"initContainers"`
```

Comment:
`InitContainers` and `Containers` are moved to the provider level. 
They are combined and filled into `Containers` with `Init bool` properly set.